myworld1/web_scrapper.py

```diff
@@ -0,0 +1,95 @@
+import psycopg2
+import requests
+import re
+from bs4 import BeautifulSoup, element
+import os
+        
+# For the credentials mentioned below, you may refer the docker-compose.yml present in myworld .
+db_name = 'member_db'
+db_user = 'postgres'
+db_pass = '123456'
+db_host = 'psql-db'
+db_port = '5432'
+        
+# This will create the connection the to postgres database.
+conn = psycopg2.connect(dbname=db_name, user=db_user, password=db_pass, host=db_host, port=db_port)
+                   
+                    
+def add_row_to_blog(title, author, date, time,content,recommended,path):
+    # This function will add the entry to database
+    sql = """INSERT INTO members_blog (title, release_date, blog_time, created_date,content, author,recommended,path) VALUES (%s, %s::DATE, %s::TIME,  NOW(),%s,%s,%s,%s)"""
+            
+    with conn:
+        with conn.cursor() as curs:
+            time=time.replace('\u202f',"")
+            curs.execute(sql, (title, date, time, content,author,recommended,path))
+        
+        
+def truncate_table():
+    # This function will delete the existing entries from the database.
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute("TRUNCATE members_blog CASCADE;")
+        
+        
+def start_extraction():
+    print("Extraction started")
+    url = "https://blog.python.org/"
+            
+    # Each time when we add new entry we delete the existing entries.
+    truncate_table()
+    data = requests.get(url)
+    page_soup = BeautifulSoup(data.text, 'html.parser')
+            
+    # Getting all the articles
+    blogs = page_soup.select('div.date-outer')
+    
+    fp=open("scrapped_data.html","w")
+    path=os.path.abspath("scrapped_data.html")
+   
+    for blog in blogs:
```

Comment:
@AnkitaHegde If any exception occurs in this loop, what happens to the file?

Reply:
As to my knowledge file will get closed automatically when exception occurs. And even some resource on internet said so and they also even suggested to use ### with as a precaution.